File: C++/1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> a;
        vector<int> b(2, -1);

        int i;
        for(i=0; i<nums.size(); i++){
            if(a.count(target-nums[i]) > 0){
                b[0] = a[target-nums[i]];
                b[1] = i;
            }
            a[nums[i]] = i;
        }

        return b;
    }
};
// @lc code=end

File: python/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        length = len(nums)
        for i in range(length-1):
            for j in range(i+1, length):
                if nums[i] + nums[j] == target:
                    return [i, j]
# @lc code=end

